dev-python/stripe/stripe-2.10.1.ebuild

```diff
@@ -0,0 +1,62 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python2_7 python3_{4,5,6,7} pypy{,3} )
+inherit distutils-r1
+
+DESCRIPTION="Stripe python bindings"
+HOMEPAGE="https://github.com/stripe/stripe-python"
+SRC_URI="mirror://pypi/s/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+
+RDEPEND="
+	$(python_gen_cond_dep '>=dev-python/requests-2[${PYTHON_USEDEP}]' 'python3*' pypy3)
+	$(python_gen_cond_dep '>=dev-python/requests-2[ssl,${PYTHON_USEDEP}]' 'python2*' pypy)
+"
+# See https://github.com/stripe/stripe-python/blob/v2.10.1/tests/conftest.py#L17
+# for minimum required version of stripe-mock
+DEPEND="${RDEPEND}
+	test? (
+		>=dev-util/stripe-mock-0.33.0
+		dev-python/pytest[${PYTHON_USEDEP}]
+		dev-python/pytest-mock[${PYTHON_USEDEP}]
+	)
+"
+
+DOCS="LONG_DESCRIPTION.rst CHANGELOG.md README.md"
+
+python_test() {
+	declare -i stripe_mock_port=12111
+	local stripe_mock_logfile="${T}/stripe_mock_${EPYTHON}.log"
+	# Try to start stripe-mock until we find a free port
+	while true; do
+		ebegin "Trying to start stripe-mock on port ${stripe_mock_port}"
+		/var/tmp/portage/dev-util/stripe-mock-0.34.1/image/usr/bin/stripe-mock --http-port ${stripe_mock_port} \
+			&> ${stripe_mock_logfile} &
+		local stripe_mock_pid=$(echo $!)
+		sleep 2
+		# Did stripe-mock start?
+		if ps -p ${stripe_mock_pid} &> /dev/null; then
```

Comment:
This looks like a horribly bad idea, and totally subject to some other process taking over the PID.

Reply:
You are saying that `$!` could return the PID of a different process due to a race condition.

In this case, I don't know how to kill `stripe-mock` at the end of the tests. The authors of the `pymongo` ebuild use `mongod --dbpath xyz --shutdown`. The mock server here does not provide any such option. Nor does it provide the option to write a pidfile. So I don't known how to get a handle for shutting down the process. Any ideas?

Reply:
No. What I'm saying is that if the process crashes, the same PID could be taken by another process. We're generally protecting against possibly dangerous behavior here but still, it would be nice to have a better way of determining whether it's running than wait-then-probe. Maybe issue some HTTP request? (wget/curl/whatever)

Reply:
Didn't even think of this possibility, but it totally makes sense. Issuing a health check via HTTP should be possible. I'll look into it, thanks!

Reply:
I am now using curl to send an authenticated request. The ebuild checks whether it receives a successful HTTP response.